lib/trailblazer/activity/dsl/linear/normalizer.rb

```diff
@@ -246,9 +246,11 @@ def inherit_option((ctx, flow_options), *)
             index = Linear::Insert.find_index(sequence, id)
             row   = sequence[index] # from this row we're inheriting options.
 
-            extensions = (row[3][:extensions]||[]) + (ctx[:extensions]||[]) # FIXME: DEFAULTING, FOR FUCK'S SAKE
+            # {inherit} all connections from {parent} step except those which are not managed by {child}.
+            connections = ctx[:outputs] ? row[3][:connections].slice(*ctx[:outputs].keys) : row[3][:connections]
```

Comment:
Can we have one private method for the heavy logic here? :muscle:

Reply:
Added it 🍻